### PR TITLE
Ensure tool_id is null for subworkflow steps

### PR DESCRIPTION
**Pull request by mvdbeek**
I contemplated dropping the key altogether, but that might have a bigger impact.

## How to test the changes?
(Select all options that apply)
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

**Comment by mvdbeek** [comment]
Errors are unrelated.

File: lib/galaxy/managers/workflows.py [hunk 1]
```diff
@@ -1337,6 +1336,8 @@ def _workflow_to_dict_export(self, trans, stored=None, workflow=None, internal=F
                 "label": step.label or None,
                 "annotation": annotation_str,
             }
+            if step.type == "tool":
+                step_dict["tool_id"] = step.tool_id
```
Comment (nsoranzo) [inline lib/galaxy/managers/workflows.py]: This looks safer:
```suggestion
                step_dict["tool_id"] = content_id
```

Comment (mvdbeek) [reply lib/galaxy/managers/workflows.py]: I disagree, use of `content_id ` is what got us here. `tool_id` is explicit and and exactly the same as the first branch of the content_id property.

Comment (mvdbeek) [reply lib/galaxy/managers/workflows.py]: We'd rewrite this for pydantic anyway and use the orm_mode, so that's also closer to what we'd do there.

Comment (nsoranzo) [reply lib/galaxy/managers/workflows.py]: What I mean is that `ToolModule.get_content_id()` [returns](https://github.com/galaxyproject/galaxy/blob/9efb749830b2331b16cb30d88e4ff08dded49e4e/lib/galaxy/workflow/modules.py#L1573) `self.tool.id if self.tool else self.tool_id` which is not always the same as `step.tool_id` .

Comment (mvdbeek) [reply lib/galaxy/managers/workflows.py]: That's not correct for exports.

**Review by nsoranzo — APPROVED** [review]

**Comment by mvdbeek** [comment]
Well, that was meant to go to 22.05 😆 ...

**Pull request merged**